### ISSUE TITLE
onCancelButtonPress is not working.

### ISSUE THREAD

**Issue by dragfire** [umhan35/react-native-search-bar#97]
Cancel button does close the keyboard but it is not calling the handler.

``` javascript
<SearchBar
                ref={ref=>this.search = ref}
                placeholder={placeholder}
                enablesReturnKeyAutomatically={false}
                searchBarStyle={'prominent'}
                onSearchButtonPress={()=> {
                    this.search.unFocus();
                }}
                onCancelButtonPress={()=> {
                    console.log('Cancel pressed');
                    this.search.unFocus();
                }}
            />
```

**Comment by iRoachie** [comment]
Is this project no longer being maintained?

**Comment by satyrius** [comment]
I have the same issue

**Comment by iRoachie** [comment]
I figured out a way to do it. Try setting up the ref this way. In my example i'm binding the focus to the state.

```jsx
<SearchBar
  ref='searchBar'
  placeholder='Search'
  showsCancelButton={this.state.focus}
  onFocus={() => this.setState({focus: true})}
  onChangeText={this.searchChanged.bind(this)}
  onSearchButtonPress={() => this.refs.searchBar.unFocus()}
  onCancelButtonPress={() => this.refs.searchBar.unFocus()}
/>
```

**Comment by gilra2000** [comment]
iRoachie// good!!

**Comment by joshuapinter** [comment]
I have `onCancelButtonPress={ () => { console.log('hi there') } }` and I find that the callback isn't even being called. And my keyboard remains up after hitting the cancel button and the SearchBar is still focused.

Are you guys finding the same thing? Any way around that? 

Thanks!

**Comment by iRoachie** [comment]
Hey guys this should be fixed in the latest release! Just published [v3.1.0](https://github.com/umhan35/react-native-search-bar/releases/tag/v3.1.0) 🎉

**Comment by bathach95** [comment]
this happened again on v 3.5.1
